NBitcoin/Utils.cs

```diff
@@ -125,6 +125,16 @@ public static IHDKey[] Derive(this IHDKey hdkey, KeyPath[] keyPaths)
 		}
 		public static async Task WithCancellation(this Task task, CancellationToken cancellationToken)
 		{
+#if NET6_0_OR_GREATER
+			try
+			{
+				await task.WaitAsync(cancellationToken).ConfigureAwait(false);
+			}
+			catch (TimeoutException e)

```

Comment:
why are you catching this? `WaitAsync` doesn't mention it

Reply:
You are right. I overlooked this.